### PR TITLE
[SPARK-44488][SQL] Support deserializing long types when creating `Metadata` object from JObject

### PR DESCRIPTION
**Pull request by richardc-db**
### What changes were proposed in this pull request?

Adds support to deserialize long types when creating `Metadata` objects from `JObject`s.

### Why are the changes needed?

Code will previously crash when adding a `long` type to the `Metadata` object.

### Does this PR introduce _any_ user-facing change?

no

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/types/Metadata.scala [hunk 1]
```diff
@@ -137,6 +137,8 @@ object Metadata {
     jObj.obj.foreach {
       case (key, JInt(value)) =>
         builder.putLong(key, value.toLong)
+      case (key, JLong(value)) =>
+        builder.putLong(key, value.toLong)
       case (key, JDouble(value)) =>
         builder.putDouble(key, value)
       case (key, JBool(value)) =>
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/types/Metadata.scala:144]: There's another case below at Array. should we fix that too?

Comment (richardc-db) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/types/Metadata.scala]: Sure, added a case for Longs in

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by scottsand-db** [comment]
Will this make Apache Spark 3.6 release? Or 4.0?

**Comment by HyukjinKwon** [comment]
It will be available from 4.0.0 most likely.

**Pull request closed without merging**